### PR TITLE
W7-project-happy-thoughts-vite-Yia

### PR DESCRIPTION
**Pull request by Citronskal**
## Netlify link
https://yia-happy-thought.netlify.app/
## Collaborators
solo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "moment": "^2.30.1",
```
Comment (ohitsnathalie) [inline package.json:13]: yes, you used moment as well 😄

File: src/assets/components/CreateThought/CreateThought.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      console.log("New thought:", newThought);
+      setMessage("");
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+
+  const handleChange = (event) => {
+    setMessage(event.target.value);
+    setError("");
```
Comment (ohitsnathalie) [inline src/assets/components/CreateThought/CreateThought.jsx:37]: clever, you thought of the error message when sending an empty message.

File: src/assets/components/ThoughtFeed/ThoughtFeed.css [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+.thought-card {
+  width: 100%;
+  max-width: 520px;
+  background-color: #ffffff;
+  border: 1px solid rgb(105, 105, 105);
+  box-shadow: 8px 8px 0 0 rgba(0, 0, 0, 1);
+  padding: 20px;
+  overflow-wrap: break-word;
```
Comment (ohitsnathalie) [inline src/assets/components/ThoughtFeed/ThoughtFeed.css:21]: good! you also figured it out after someone wrote a long "word" as a thought 😄

File: src/assets/components/ThoughtFeed/ThoughtFeed.css [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    padding: 8px;
+    font-size: 16px;
+  }
+
+  .post-time {
+    font-size: 12px;
+  }
+}
+
+/* Mobile version */
```
Comment (ohitsnathalie) [inline src/assets/components/ThoughtFeed/ThoughtFeed.css:77]: have you tried to do mobile first and then go for the bigger screens?

File: src/assets/components/CreateThought/CreateThought.jsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import { useState } from "react";
+import "./CreateThought.css"; 
+
+const APIURL = "https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts";
```
Comment (ohitsnathalie) [inline src/assets/components/CreateThought/CreateThought.jsx:4]: you already defined the APIURL in the ThoughtFeed.jsx. You could have passed the APIURL to this component.

File: src/assets/components/ThoughtFeed/ThoughtFeed.jsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      const json = await response.json();
+      setThoughts(json);
+    } catch (error) {
+      console.error("Error fetching thoughts:", error);
+    }
+  };
+
+  const handleLike = async (thoughtId) => {
+    try {
+      const response = await fetch(`${APIURL}/${thoughtId}/like`, {
```
Comment (ohitsnathalie) [inline src/assets/components/ThoughtFeed/ThoughtFeed.jsx:30]: interesting way to change the APIURL to the "liked" version. Cool to see a different approach.

**Review by ohitsnathalie — COMMENTED** [review]
Great job Yia!
Your Happy Thoughts project looks like the given styling and also is responsive on different devices.
I realized that when I posted a thought, it doesn't automatically show up. After I refreshed the page, my message was there.
Try to write your README.md so we can see where you had some troubles. Also, try another approach with responsiveness and start with the mobile version first. Maybe you can tell a difference. 
Overall, I like what you did and you can be proud of your project ❤️

**Review by dzc1 — APPROVED** [review]
Hey Yia! great project 💪💪

Loved the folder structur, all requiirements have been met and it seems that you are even workig with moment.js which to put simply, I can see that you'r learnning curve is exponential and you are digging the framework. Now, I'll give you the passing grade here Yia but let's make a [deal! ](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExcjNmbWFtdDdreDBwZmVhaWVzM2Q4eG53a29qamoxOWF1NzBtZ3Z1biZlcD12MV9naWZzX3NlYXJjaCZjdD1n/jZAOQbch4ibfQ97VBS/giphy.gif). 

Some knowledge for some commitment! 

Basically, everytiime you post a new thought on your app, you have t manually refresh the browser which cool but who has [time for that right](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExb2l6NzRhN2s3bGIxc3pyc2x6Ynd2eWxtMmNoMm1kNzV3OGVkaWVwNSZlcD12MV9naWZzX3NlYXJjaCZjdD1n/XBEoaajXTXaALzawSn/giphy.gif)?  Ennter the useEffect to the rescue. 

TO solve thiis, we can sue the dependency array in your `ThoughtFeed.jsx` file specifically on line 13. 

Change it form this: 

 ```
useEffect(() => {
    fetchData();
  }, []);
```

to this 

 ```
useEffect(() => {
    fetchData();
  }, [recentThoughts]);
```

This will cover the use-case of the hard refresh and you will take ful advantage of the useEffect hook! 
 
 So, do we have a[ deal](https://media.giphy.com/media/vV5Jcf6yQsQPI0p4WH/giphy.gif?cid=790b7611buiqfztcee6r0lo8qbv3it1mez1vtq1ksn5xmwry&ep=v1_gifs_search&rid=giphy.gif&ct=g) ? 
 
 Keep up the great work Yiia! hope that you are enjoying the react framework

**Pull request closed without merging**